### PR TITLE
Add Iron165R2 PCB support

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Adds support for Iron165 R2 PCB. Both this and the Iron180 PCBs were moved to a `smithrune` folder inside `keyboards`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/smithrune/iron165r2/chconf.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
```
Comment (drashna) [inline keyboards/smithrune/iron165r2/chconf.h:17]: Is this config file necessary, or just copied?

File: keyboards/smithrune/iron180/chconf.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * This file was auto-generated by:
+ *    `qmk chibios-confmigrate -i keyboards/iron180/chconf.h -r platforms/chibios/common/configs/chconf.h`
+ */
+
+#pragma once
```
Comment (drashna) [inline keyboards/smithrune/iron180/chconf.h:22]: Is this config actually needed?

File: keyboards/smithrune/iron180/config.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+//#define NO_ACTION_MACRO
+//#define NO_ACTION_FUNCTION
```
Comment (drashna) [inline keyboards/smithrune/iron180/config.h:70]:
```suggestion
```

File: keyboards/smithrune/iron180/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
```
Comment (drashna) [inline keyboards/smithrune/iron180/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/smithrune/iron180/rules.mk [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/smithrune/iron180/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/smithrune/iron165r2/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+#
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/smithrune/iron165r2/rules.mk:15]:
```suggestion
NKRO_ENABLE = yes            # Enable N-Key Rollover
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/smithrune/iron180/chconf.h (hunk 2 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron180/chconf.h:17]: To be honest it was already there, I just copied it -- let me see if it the firmware works fine without it

File: keyboards/smithrune/iron165r2/chconf.h (hunk 1 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron165r2/chconf.h:17]: Copied, I thought it was necessary. Let me get a PCB and test if the firmware works without it

File: keyboards/smithrune/iron180/config.h (hunk 3 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron180/config.h]: Committed in https://github.com/qmk/qmk_firmware/pull/16948/commits/75e44149f2d4e80b919039df66960d4f6b997d66

File: keyboards/smithrune/iron180/rules.mk (hunk 4 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron180/rules.mk]: Committed in https://github.com/qmk/qmk_firmware/pull/16948/commits/4dd310000ad52978119097e0a08d134f4c695db2

File: keyboards/smithrune/iron165r2/rules.mk (hunk 6 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron165r2/rules.mk]: Committed in https://github.com/qmk/qmk_firmware/pull/16948/commits/cb53f66b3904fca0f1aba49a7c3397fad98acda5

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/smithrune/iron165r2/readme.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+* **Keycode in layout**: Press the key mapped to `RESET`; in the default layout, that is top left key ('escape') in layer 1.
+
+### How to compile and flash
+
+The 165R2 PCB was prototyped and tested using two microcontrollers: STM32F072 and STM32F411. Most production-run units were manufactured with the latter. 
+
+After checking what PCB you own and setting up your build environment, you can compile the Iron 165R2 default keymap by using one of the options below.
+
+    make smithrune/iron165r2/165r2_f072:default
+    make smithrune/iron165r2/165r2_f411:default
```
Comment (GreatWizard) [inline keyboards/smithrune/iron165r2/readme.md:30]:
```suggestion
    make smithrune/iron165r2/f411:default
```

File: keyboards/smithrune/iron165r2/readme.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+And use dfu-util in the command line or through a GUI like QMK toolbox to upload the firmware to the PCB. To directly flash the PCB after it is put into a DFU state, use:
+
+    make smithrune/iron165r2/165r2_f072:default:flash
+    make smithrune/iron165r2/165r2_f411:default:flash
```
Comment (GreatWizard) [inline keyboards/smithrune/iron165r2/readme.md:35]:
```suggestion
    make smithrune/iron165r2/f411:default
```

File: keyboards/smithrune/iron165r2/readme.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+The 165R2 PCB was prototyped and tested using two microcontrollers: STM32F072 and STM32F411. Most production-run units were manufactured with the latter. 
+
+After checking what PCB you own and setting up your build environment, you can compile the Iron 165R2 default keymap by using one of the options below.
+
+    make smithrune/iron165r2/165r2_f072:default
+    make smithrune/iron165r2/165r2_f411:default
+
+And use dfu-util in the command line or through a GUI like QMK toolbox to upload the firmware to the PCB. To directly flash the PCB after it is put into a DFU state, use:
+
+    make smithrune/iron165r2/165r2_f072:default:flash
```
Comment (GreatWizard) [inline keyboards/smithrune/iron165r2/readme.md:34]:
```suggestion
    make smithrune/iron165r2/f072:default
```

**Comment by GreatWizard** [comment]
Is there any reason to keep https://github.com/qmk/qmk_firmware/blob/master/keyboards/iron180/readme.md ? as it's the same now in smithrune directory

**Pull request merged**